lib/waterwheel.js

```diff
@@ -126,4 +126,30 @@ module.exports = class Waterwheel extends Request {
       });
   }
 
+  /**
+   * Fetch embedded resources from HAL+JSON documents
+   * @param  {object} entityJSON
+   *   An object, usually returned from Drupal, containing _embedded information
+   * @return {Promise}
+   *   If no _embedded key is found, a rejection is returned, else a resolved
+   *   promise with all the embedded resources requests completed.
+   */
+  fetchEmbedded(entityJSON) {
+    if (!entityJSON || !entityJSON.hasOwnProperty('_embedded')) {
+      return Promise.reject('This is probably not HAL+JSON');
+    }
+
+    let links = [];
+
+    Object.keys(entityJSON._embedded).forEach(key => {
+      entityJSON._embedded[key].forEach(ref => {
+        links.push(ref._links.self.href.split(this.base)[1]);
+      });
+    });
+
+    links = Array.from(new Set(links));
+
+    return Promise.all([Promise.resolve(entityJSON)].concat(links.map(link => this.issueRequest(methods.get, link))));
```

Comment:
I'm assuming your doing this so that the parent entity is provided adjacent to the embedded ones? A comment could help make the intent here clear.